localstack/services/stepfunctions/asl/component/state/state_execution/state_task/service/resource.py

```diff
@@ -85,10 +84,12 @@ def __init__(self, resource_arn: ResourceARN):
         self.account = resource_arn.account
 
     @staticmethod
-    def from_resource_arn(arn: str) -> Resource:
+    def from_resource_arn(account_id: str, region_name: str, arn: str) -> Resource:
```

Comment:
Runtime values are ideally evaluated during execution. So resource should become an EvalComponent, evaluate its region and account in `_eval_body` using the environment, and push to the stack a resource object that includes all the values.